docs/rtk-query/api/created-api/miscellaneous-utils.mdx

```diff
@@ -0,0 +1,61 @@
+---
+id: miscellaneous-utils
+title: 'API Slices: Miscellaneous Utils'
+sidebar_label: Miscellaneous Utils
+hide_title: true
+---
+
+&nbsp;
+
+# API Slices: Miscellaneous Utilities
+
+The API slice object includes miscellaneous utilities that may be used for scenarios such as SSR, Suspense for Data Fetching, etc.
+
+:::info Usage Reference
+
+- [Server Side Rendering](../../usage/server-side-rendering.mdx)
+
+:::
+
+Each API slice object has `getRunningOperationPromises` and `getRunningOperationPromise` functions.
+
+## `getRunningOperationPromises`
+
+#### Signature
+
+```ts no-transpile
+const getRunningOperationPromises: () => Array<Promise<unknown>>
+```
+
+#### Description
+
+A function that returns all promises for running queries and mutations.
+
+This is useful for SSR scenarios to await everything triggered in any way, including via hook calls,
+or manually dispatching `initiate` actions.
+
+```ts no-transpile title="Awaiting all currently running queries & mutations example"
+await Promise.all(api.getRunningOperationPromises())
+```
+
+## `getRunningOperationPromise`
+
+#### Signature
+
+```ts no-transpile
+const getRunningOperationPromise: <EndpointName extends QueryKeys<Definitions>>(
+  endpointName: EndpointName,
+  args: QueryArgFrom<Definitions[EndpointName]>
+) =>
+  | QueryActionCreatorResult<Definitions[EndpointName]>
+  | MutationActionCreatorResult<Definitions[EndpointName]>
+  | undefined
+```
+
+#### Description
+
+A function that returns a single promise for a given endpoint name + argument combination,
+if it is currently running. If it is not currently running, the function returns `undefined`.
+
+This enables writing custom hooks that look up if RTK Query has already got a running promise
+for a certain endpoint/argument combination, and retrieving that promise to `throw` it.
```

Comment:
Note that I didn't explicitly mention `suspense` here, given it's experimental nature.

Is it worth keeping the line about 'enabling writing custom hooks' without any context as to *why* you would write it?

Or should we word it somehow to note that it may allow for usage with `suspense` in the future?